cl/corpus_importer/management/commands/update_casenames_wl_dataset.py

```diff
@@ -0,0 +1,360 @@
+        # if no V. - likely an "in re" case and only match on at least 1 name
+        return True
+
+    # otherwise check if a match occurs on both sides of the V
+    v_index = csv_case_name.lower().index("v.")
+    hit_indices = [csv_case_name.lower().find(hit) for hit in overlap]
+
+    return min(hit_indices) < v_index < max(hit_indices)
+
+
+def parse_date(date_str: str) -> date | None:
+    """Attempts to parse the filed date into a datetime object.
+
+    January 10, 1999
+    24-Jul-97
+    21-Jan-94
+    1/17/1961
+    12/1/1960
+    26-Sep-00
+    Feb. 28, 2001
+
+    :param date_str: date string
+    :return: date object or none
+    """
+    for fmt in DATE_FORMATS:
+        try:
+            return datetime.strptime(date_str, fmt).date()
+        except (ValueError, TypeError):
+            continue
+    logger.warning(f"Invalid date format: {date_str}")
+    return None
+
+
+def parse_citations(citation_strings: list[str]) -> list[dict]:
+    """Validate citations with Eyecite.
+
+    :param citation_strings: List of citation strings to validate.
+    :return: List of validated citation dictionaries with volume, reporter, and page.
+    """
+    validated_citations = []
+
+    for cite_str in citation_strings:
+        # Get citations from the string
+        found_cites = get_citations(cite_str, tokenizer=HYPERSCAN_TOKENIZER)
+
+        # Ensure we have valid citations to process
+        for citation in found_cites:
+            if isinstance(citation, FullCaseCitation):
+                volume = citation.groups.get("volume")
+
+                # Validate the volume
+                if volume and volume.isdigit():
+                    # Append the validated citation as a dictionary
+                    validated_citations.append(
+                        {
+                            "volume": citation.groups["volume"],
+                            "reporter": citation.corrected_reporter(),
+                            "page": citation.groups["page"],
+                        }
+                    )
+                else:
+                    # If volume is invalid, skip this citation
+                    continue
+
+    return validated_citations
+
+
+def query_possible_matches(
+    valid_citations: list[dict], docket_number: str, date_filed: date
+) -> QuerySet[Citation]:
+    """Find matches for row data
+
+    :param valid_citations: list of FullCaseCitation objects
+    :param docket_number: cleaned docket number from row
+    :param date_filed: formatted filed date from row
+
+    :return: list of matched OpinionCluster
+    """
+
+    citation_queries = Q()
+
+    for citation in valid_citations:
+        citation_query = Q(**citation) & Q(
+            cluster__docket__docket_number__contains=docket_number,
+            cluster__date_filed=date_filed,
+        )
+        citation_queries |= citation_query
+    possible_matches = Citation.objects.filter(
+        citation_queries
+    ).select_related("cluster")
+
```

Comment:
if there are no valid citations - what happens?  Will this just return a query of all citations?  that would be bad right?